sources/api/migration/migrations/v1.18.1/corndog-config-file-v0-1-0/src/main.rs

```diff
@@ -0,0 +1,30 @@
+use migration_helpers::common_migrations::{
+    AddPrefixesMigration, ListReplacement, ReplaceListsMigration,
+};
+use migration_helpers::{migrate, Result};
+use std::process;
+
+fn run() -> Result<()> {
+    migrate(AddPrefixesMigration(vec![
+        "configuration-files.corndog-toml",
+    ]))?;
+    migrate(ReplaceListsMigration(vec![
+        ListReplacement {
+            setting: "services.sysctl.configuration-files",
+            old_vals: &[],
+            new_vals: &["corndog-toml"],
+        },
+        ListReplacement {
+            setting: "services.lockdown.configuration-files",
+            old_vals: &[],
+            new_vals: &["corndog-toml"],
+        },
+    ]))
```

Comment:
Are there other migrations where we make two `migrate` calls in the same binary? I seem to recall that didn't work correctly - that we end up losing the results of all but the last migration.

For the "Add" family the truly important actions only happen on downgrade; they exist to clean up files the old version doesn't expect. Upgrade is a no-op because storewolf will create new entries.

For the ListReplace steps both directions are important.

Reply:
I checked the other migrations and the only place we've made multiple `migrate` calls is when they are conditionally compiled.

The way that migrations are implemented, we don't get a new copy of the data store that reflects our changes until the migration binary exits. So if the function is run a second time, it starts from original input state again.